lib/src/core/engine.dart

```diff
@@ -353,39 +365,30 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
       subscriber?.pc.onDataChannel = _onDataChannel;
     }
 
-    subscriber?.pc.onIceConnectionState =
-        (state) => events.emit(EngineSubscriberIceStateUpdatedEvent(
-              iceState: state,
+    subscriber?.pc.onConnectionState =
+        (state) => events.emit(EngineSubscriberPeerStateUpdatedEvent(
+              state: state,
               isPrimary: _subscriberPrimary,
             ));
 
-    publisher?.pc.onIceConnectionState =
-        (state) => events.emit(EnginePublisherIceStateUpdatedEvent(
+    publisher?.pc.onConnectionState =
+        (state) => events.emit(EnginePublisherPeerStateUpdatedEvent(
               state: state,
               isPrimary: !_subscriberPrimary,
             ));
 
-    events.on<EngineIceStateUpdatedEvent>((event) {
+    events.on<EnginePeerStateUpdatedEvent>((event) {
       // only listen to primary ice events
       if (!event.isPrimary) return;
 
-      if (event.iceState ==
-          rtc.RTCIceConnectionState.RTCIceConnectionStateConnected) {
-        if (!_iceConnected) {
-          _iceConnected = true;
-          if (_connectionState == ConnectionState.reconnecting) {
-            events.emit(const EngineReconnectedEvent());
-          } else {
-            events.emit(const EngineConnectedEvent());
-          }
-        }
-      } else if (event.iceState ==
-          rtc.RTCIceConnectionState.RTCIceConnectionStateFailed) {
+      // if (event.iceState ==
```

Comment:
If code is not used, can we delete it? Just to avoid any confusion for future engineers :-)